examples/citi_bike/online_lp/citi_bike_lp.py

```diff
@@ -0,0 +1,187 @@
+        self._station_capacity = station_capacity
+        self._station_neighbor_list = station_neighbor_list
+        self._decision_interval = decision_interval
+
+        self._plan_window_size = config.plan_window_size
+        self._apply_buffer_size = config.apply_buffer_size
+        self._safety_inventory_limit = config.safety_inventory_limit
+
+        self._fulfillment_time_decay_factor = config.objective.fulfillment_time_decay_factor
+        self._transfer_cost_factor = config.objective.transfer_cost_factor
+        self._safety_inventory_reward_factor = config.objective.safety_inventory_reward_factor
+
+        self._num_tick = math.ceil(self._plan_window_size / self._decision_interval)
+        self._last_start_tick = -1
+
+    def _init_variables(self, init_inventory: np.ndarray):
+        self._inventory = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
+        self._safety_inventory = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
+        self._fulfillment = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
+        self._transfer = np.zeros((self._num_tick, self._num_station, self._num_neighbor), dtype=np.int32).tolist()
+
+        # The intermediate variables for clearer constraints.
+        self._transfer_from = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
+        self._transfer_to = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
+
+        for tick in range(self._num_tick):
+            for station in range(self._num_station):
+                self._inventory[tick][station] = LpVariable(
+                    name=f"T{tick}_S{station}_Inv", lowBound=0, upBound=self._station_capacity[station], cat="Integer"
+                )
+                self._safety_inventory[tick][station] = LpVariable(
+                    name=f"T{tick}_S{station}_SafetyInv",
+                    lowBound=0,
+                    upBound=round(self._safety_inventory_limit * self._station_capacity[station]),
+                    cat="Integer"
+                )
+                self._fulfillment[tick][station] = LpVariable(
+                    name=f"T{tick}_S{station}_Fulfillment", lowBound=0, cat="Integer"
+                )
+
+                # For intermediate variables.
+                self._transfer_from[tick][station] = LpVariable(
+                    name=f"T{tick}_TransferFrom{station}", lowBound=0, cat="Integer"
+                )
+                self._transfer_to[tick][station] = LpVariable(
+                    name=f"T{tick}_TransferTo{station}", lowBound=0, cat="Integer"
+                )
+
+                for neighbor_idx in range(self._num_neighbor):
+                    self._transfer[tick][station][neighbor_idx] = LpVariable(
+                        name=f"T{tick}_Transfer_from{station}_to{neighbor_idx}th", lowBound=0, cat="Integer"
+                    )
+
+        # Initialize the current inventory.
```

Comment:
Initialize the first decision tick inventory with the environment's current inventory.